### PR TITLE
Split files vuex into multiple files for better maintainability

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
The store is already quite big and it will get even bigger (sharing, file versions ....)
With this we are following best practive as described in https://vuex.vuejs.org/guide/structure.html

## How Has This Been Tested?
- :hand: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [x] file list is cleared after opening the side bar :confused: 
- [x] results number in files bar is not displayed

### CODE REVIEW

**Comment by ownclouders** [comment]
Find screenshots at https://minio.owncloud.com/phoenix/screenshots/2258/

**Comment by LukasHirt** [comment]
Compilation throws errors and doesn't start up the local server.

image

File: apps/files/src/store/getters.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      for (let filter of state.fileFilter) {
+        if (file.type === filter.tag) {
+          if (!filter.value) return false
+        } else if (file.name.startsWith('.')) {
+          // show hidden files ?
+          if (!state.fileFilter[2].value) return false
+        }
+      }
+      // respect filename filter for local 'search' in open folder
+      return !(state.searchTermFilter && !file.name.toLowerCase().includes(state.searchTermFilter.toLowerCase()));
```
Comment (LukasHirt) [inline apps/files/src/store/getters.js]: Also, this semicolon throws an error while compiling.

File: apps/files/src/store/index.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import mutations from './mutations'
+import getters from './getters'
+const namespaced = true
+
+export default {
+  namespaced,
+  state,
+  getters,
+  actions,
+  mutations,
```
Comment (LukasHirt) [inline apps/files/src/store/index.js]: This trailing comma also throws an error

**Comment by LukasHirt** [comment]
> Compilation throws errors and doesn't start up the local server.
> 
> image

This is probably related to something else. I just discovered that it does the same to me even at the master.

**Pull request merged**